src/animations.ts

```diff
@@ -111,105 +455,107 @@ let FAILED_KARAOKE_UPDATE_ATTEMPTS = 0
+        karaokeState = attemptUpdateKaraoke(animation.karaoke, animationTime, karaokeState)
+        window.requestAnimationFrame(() => sampleQuestionAnimation(animationName, karaokeState))
+    } else if (!isSameAudio) {
+        // Another audio file has started playing, so this animation can be reset.
+        animation.cleanupAnimation()
+    }
 }
 
 
-function yourLittleOneAnimation() {
-    if (CURRENT_ANIMATION_INFO.name === YOUR_LITTLE_ONE_ANIMATION) {
-        const animation = ANIMATIONS[YOUR_LITTLE_ONE_ANIMATION]
-        const isAudioPlaying = !PLAYER.paused && (PLAYER.querySelector('source').src === animation.expectedAudioSrc)
-
-        // animationTime represents the current progress into the animation.
-        // If the appropriate audio is playing, animationTime is the number of milliseconds from the player's
-        // currentTime.
-        // Otherwise, animationTime is the number of milliseconds since the animation was scrolled into view (and using
-        // the modulo operator to account for looping animations).
-        const animationTime = (
-            isAudioPlaying ?
-                PLAYER.currentTime * 1000 :
-                ((new Date()).valueOf() - CURRENT_ANIMATION_INFO.timeScrolledIntoView) % animation.duration
-        )
-
-        for (const step of animation.steps) {
-            const element = document.getElementById(step.id)
-            if (animationTime >= step.start && animationTime < step.end) {
-                element.style.opacity = '1'
-            } else {
-                element.style.opacity = '0'
-            }
-        }
+// Set the animation back to the initial state.
+function sampleQuestionAnimationCleanup(animationName: AnimationName) {
+    const animation = ANIMATIONS[animationName]
+    setRadialProgressBar(animation, 0)
+    attemptUpdateKaraoke(animation.karaoke, 0, null)
+}
 
-        setRadialProgressBar(animation, animationTime)
-        attemptUpdateKaraoke(animation.karaoke, animationTime)
 
-        window.requestAnimationFrame(yourLittleOneAnimation)
-    }
-}
+let FAILED_GET_SAMPLE_QUESTION_COMPONENTS_ATTEMPTS = 0
+
+// Since the sample questions are created programmatically from the CMS, they may not be in the DOM by the time this
+// script runs (even with `defer`). Attempt to fetch the components for 10 seconds before sending a Sentry alert.
+function getSampleQuestionComponents() {
+    const sampleQuestionsSlider = $('.sample-questions-slider')
+    for (const animationName of SAMPLE_QUESTION_ANIMATIONS) {
+        // The components for these animations are programmatically created from the CMS collection, so they cannot be
+        // reliably found with a class or id. Use the audio URL to find the slide associated with this animation.
+        const slide = sampleQuestionsSlider.find(
+            `div[data-element='url']:contains('${ANIMATIONS[animationName].expectedAudioSrc}')`
+        ).closest('.sample-questions-slide')
 
+        if (slide.length) {
```

Comment:
Is it risky not to check that `slide` is defined first (to prevent errors like "undefined is not an object" when trying to access `.length`)? Are we certain that the selector above will always match a target? Not to crash

Reply:
Good question! Since `slide` is retrieved with jQuery, even if the selectors aren't able to find anything it will still return an empty jQuery object with a length of 0